Configuration/src/main/java/io/deephaven/configuration/Configuration.java

```diff
@@ -149,14 +139,27 @@ public static void reset(final @NotNull String name) {
         NAMED_CONFIGURATIONS.remove(name);
     }
 
-    @SuppressWarnings("UnusedReturnValue")
     public static Configuration newConfigurationForTesting() {
-        final Configuration newConfig = new Configuration();
+        return newConfigurationForTesting(DEFAULT_CONF_NAME, DefaultConfigurationContext::new);
+    }
+
+    public static Configuration newConfigurationForTesting(final @NotNull String name) {
+        return newConfigurationForTesting(name, DefaultConfigurationContext::new);
+    }
+
+    @SuppressWarnings("UnusedReturnValue")
+    public static Configuration newConfigurationForTesting(final @NotNull String name,
```

Comment:
Maybe isntead of "forTesting" call it "newStandaloneConfiguration"; as it could be used for other cases that should not be in the default map and tests could still use it.

Reply:
We should also use `@TestUseOnly` or `@TestOnly` annotations.

Reply:
I think Charles' point was that we might want to use this for not testing too

Reply:
Yes, specifically I was thinking we would rename it away from "forTesting" and then we could use it in the Enterprise "dhconfig" command.  I would like to be able to create a configuration distinct from the main process configuration, with alternative parameters for the stanzas (or maybe an alternative root file) and understand what values a property takes on.

We have an old tool for that, but it is not nearly as good as the dhconfig suite and we have the potential to do much better with a slightly modified Configuration that we can make in a standalone way.